### ISSUE TITLE
Add more options for startup folding levels

### ISSUE THREAD

**Issue by gusbrs** [alphapapa/outshine#50]
This is a feature request. Org itself has more options as to the initial folded state of a buffer, namely the option `org-startup-folded` can receive values `overview`, `showall` or `content`. Outshine only offers `outshine-startup-folded-p` (and, implicitly, to start in a `showall` state). It would be nice if the equivalent of `content` could be also made available in Outshine.

Outshine uses `outshine-startup-folded-p` in the following passage of `outshine--minor-mode-activate`:

```
(when outshine-startup-folded-p
  (condition-case error-data
      (outline-hide-sublevels 1)
    ('error (message "No outline structure detected"))))
```

Thus the initial folding is being dealt with with an `outline-mode` function. But `outline-mode` already offers, besides `outline-hide-sublevels`, `outline-hide-body` and, if needed, `outline-show-all`. Which means the necessary functions are readily available.

I know the recent refactoring is drawing most of the attention as of late, but if some time could eventually be devoted to this minor improvement, it would be appreciated. By the way, thank you for the nice work being done here.

**Comment by alphapapa** [comment]
Thanks.

**Comment by alphapapa** [comment]
@yuhan0 Great!

> I don't think "content" is a particularly good word to describe showing all headlines but hiding body text, but it's probably better to follow Org conventions

I think we can both "lead the way" to better descriptions and follow convention at the same time.  How about:

    All headlines (aka "Content" in Org)

**Comment by yuhan0** [comment]
Thanks for merging! I wasn't referring to the description but the symbol `'content`, which is the keyword used in the Org variable `org-startup-folded` and `#+STARTUP` tags.

**Comment by gusbrs** [comment]
@yuhan0 Sorry, I'm probably missing something here. But I can't seem to find this particular `defcustom` either in the pull request or in the new master branch. Could you please point me in the right direction?

**Comment by alphapapa** [comment]
> Thanks for merging! I wasn't referring to the description but the symbol `'content`, which is the keyword used in the Org variable `org-startup-folded` and `#+STARTUP` tags.

Sure, but I was referring to the description.  :)

**Comment by yuhan0** [comment]
@gusburs it wasn't in that PR or on any branch on Github now, I was just referencing a change I made on my own local copyb- sorry for any confusion.
I should be publishing it soon, still figuring out the best way of dealing with dealing with the point being within a folded headline (may be okay for Org buffers, but very annoying for code where you lose the point's position)

**Comment by gusbrs** [comment]
@yuhan0 Oh, I see. I'll be looking forward to see it published then, when it is possible. Thank you very much.

Indeed, the question of the point within a folded line deserves care. I've been playing with this (for the time being I simply added `:hook ((outshine-mode . outline-hide-body) (emacs-lisp-mode . outshine-mode))`). And using `xref-find-definitions` (M-.) to an elisp buffer (given I enabled outshine for them), or clicking on a reference in an `rgrep` buffer to an elisp buffer, leaves one in an unfavourable position.

**Comment by yuhan0** [comment]
Yeah, that turns out to be a problem for plenty of commands which jump around the buffer at arbitrary positions like `imenu`, `magit-diff-visit-file`, `flycheck-jump-to-error`, etc. 

If you're interested to try it out in the meantime, here's the helper function - 
```emacs-lisp
;;;###autoload
(defun outshine-reveal-context (&optional arg)
  "Expands the folded headline around the current point, (similar to what happens
  during isearch.) Use this to advise functions which may jump to the interior of a
folded section."
  (interactive)
  (when (and outshine-mode
             (invisible-p (point)) ;; faster than checking for heading
             (not (outline-on-heading-p nil)))
    (dolist (ov (overlays-at (point)))
      (when (memq 'outline (overlay-properties ov))
        (delete-overlay ov)))
    ;; (ignore-errors (save-excursion
    ;;                  (outline-back-to-heading t)
    ;;                  (outline-show-entry)
    ;;                  (outline-show-children)))
    ))
```

I started by using it to advise a list of functions which behave in that manner, but after a while found it easier to simply add it to the post-command-hook.

```
(add-hook 'post-command-hook 'outshine-reveal-context)
```

**Comment by alphapapa** [comment]
@yuhan0 FYI that function doesn't need to be autoloaded, as it won't be called (or certainly shouldn't be) until after Outshine is already loaded.
